scripts/ci_setup.py

```diff
@@ -361,9 +367,14 @@ def __main(args: list) -> int:
             out_file.write(variable_format % ('UseSharedCompilation', 'false'))
             out_file.write(variable_format % ('DOTNET_ROOT', dotnet_path))
             out_file.write(variable_format % ('MAUI_VERSION', args.maui_version))
+            out_file.write(path_variable % dotnet_path)
             if args.affinity:
                 out_file.write(variable_format % ('PERFLAB_DATA_AFFINITY', args.affinity))
-            out_file.write(path_variable % dotnet_path)
+            if args.run_env_vars:
+                for env_var in args.run_env_vars:
+                    key, value = env_var.split('=', 1) 
+                    out_file.write(variable_format % (key, value))
+                    out_file.write(variable_format % ("PERFLAB_DATA_" + key, value))
```

Comment:
remind me, but these I believe also gets set in BDN, right? Can you check if it overrides the `HeapCountCharacteristic`?

Reply:
According to https://benchmarkdotnet.org/articles/configs/jobs.html#environment, host process environment setups are used when not specified otherwise. I am not finding anything in the BDN src that would point to otherwise, @adamsitnik if you have any insight off the top of your head it would be useful. I also looked to make sure we are not setting anything in the performance repo and did not find any references to things related to the GC in either the Test files or the benchmarks.extension. We will also verify these EnvVars locally to make sure that they seem to be having an impact before pushing them in a separate PR.

Reply:
> Host process environment setups are used when not specified otherwise

That is true.